segmentstore/server/src/main/java/io/pravega/segmentstore/server/writer/SegmentAggregator.java

```diff
@@ -468,10 +471,12 @@ private void aggregateAppendOperation(CachedStreamSegmentAppendOperation operati
         if (operation.getStreamSegmentOffset() < aggregatedAppend.getLastStreamSegmentOffset()) {
             // The given operation begins before the AggregatedAppendOperation. This is likely due to it having been
             // partially written to Storage prior to some recovery event. We must make sure we only include the part that
-            // has not yet been written.
+            // has not yet been written. Also, we trigger reconciliation to check that the partially written data is correct.
             long delta = aggregatedAppend.getLastStreamSegmentOffset() - operation.getStreamSegmentOffset();
             remainingLength -= delta;
-            log.debug("{}: Skipping {} bytes from the beginning of '{}' since it has already been partially written to Storage.", this.traceObjectId, delta, operation);
+            log.info("{}: Skipping {} bytes from the beginning of '{}' since it has already been partially written to Storage. Triggering reconciliation.",
+                    this.traceObjectId, delta, operation);
+            setState(AggregatorState.ReconciliationNeeded);
```

Comment:
@andreipaduroiu in this PR, we want to trigger reconciliation in case we find a partially written event. This is because it might allow the system to check that the data read from Bookkeeper and the one on LTS are actually the same if this happens. What is your take on this? Thanks!

Reply:
I don't see the point. The Segment Read Index takes care of unifying the LTS and Tier1/Cache data with no input from upstream layers. What are you trying to solve here? This will just slow down recoveries. Think that you may have many hundreds of segments that need to be recovered and likely all of them are in this situation.

Reply:
Thanks @andreipaduroiu. That's why I asked your input here: I think we had 2 different interpretations of what "partial writes" mean. Here, it looks like it just refers to the fact that flushes to LTS, which are fixed size (e.g., 64MB), may end with a partial append. The rest of the append would get flushed in the next `AggregatedAppend`, and at this point is where the above code path is relevant.

Our interpretation of "partial write" here was more related to the fact that SLTS could write some data partially to LTS (e.g., failure, recovery, etc.). Whatever data that could have been written to LTS, if any, was our main understanding of "partial write". Note that this was in the context of some investigation of a data corruption issue. Considering this definition of "partial write", it looked reasonable to trigger reconciliation. Hope that this clarifies the misunderstanding.

Note that the above definition of "partial write" (the one we considered), should not be possible as soon as PR #6841 is in. Thanks a lot for your feedback again!